src/core/lib/iomgr/executor.cc

```diff
@@ -264,6 +264,10 @@ void Executor::ThreadMain(void* arg) {
     grpc_core::ExecCtx::Get()->InvalidateNow();
     subtract_depth = RunClosures(ts->name, closures);
   }
+#if !defined(__APPLE__)  // because of Thread local issues with MacOS and iOS
```

Comment:
Please expand the comment into a multi-line comment explaining the issue that is being fixed here, why it can't be fixed on Apple platforms (thread-locals not being well-supported), and self-assign a TODO to fix this once we can upgrade our minimum Xcode requirement or whatever it takes.

Reply:
Done